src/components/EnhancedChatInterface.tsx

```
@@ -0,0 +1,1126 @@
+  
+  // Enhanced animations
+  const [messageAnimations, setMessageAnimations] = useState<{ [key: string]: boolean }>({});
+
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+  const messagesEndRef = useRef<HTMLDivElement>(null);
+  const inputContainerRef = useRef<HTMLDivElement>(null);
+
+  // Convex queries and mutations
+  const chats = useQuery(api.chats.getChats);
+  const messages = selectedChatId 
+    ? useQuery(api.messages.getMessages, { chatId: selectedChatId as Id<'chats'> })
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)